src/services/dashboards/widgets/_helpers/widget-helper.ts

```diff
@@ -73,3 +82,31 @@ export const getWidgetComponent = (widgetConfigId: string): AsyncComponent => {
 
     return widgetComponent;
 };
+
+export const getWidgetSchemaErrorMap = (
+    inheritOptions?: InheritOptions,
+    widgetOptionsSchema?: WidgetOptionsSchema['schema'],
+    dashboardVariables?: DashboardVariables,
+    dashboardVariablesSchema?: DashboardVariablesSchema,
+): InheritOptionsErrorMap => {
+    if (!inheritOptions || isEmpty(inheritOptions)) {
+        return {};
+    }
+    const errorMap: InheritOptionsErrorMap = {};
+    Object.entries(inheritOptions).forEach(([propertyName, inheritOption]) => {
+        if (!inheritOption?.enabled) return;
+
+        const variableKey = inheritOption?.variable_info?.key;
+        if (!variableKey || !dashboardVariablesSchema?.properties?.[variableKey] || !dashboardVariables?.[variableKey]) {
+            errorMap[propertyName] = i18n.t('This property does not exist on the dashboard variables.');
+            return;
+        }
+
+        const variableType = dashboardVariablesSchema.properties[variableKey].selection_type === 'MULTI' ? 'array' : 'string';
+        const widgetPropertyType = widgetOptionsSchema.properties[propertyName].type;
+        if (variableType !== widgetPropertyType) {
+            errorMap[propertyName] = i18n.t('This property has a different type from the dashboard variable.');
```

Comment:
suggestion: Could you add `<!-- song lang -->`annotation for translating?😀

Reply:
Oh I missed it! thanks!